source/compiler/qsc_doc_gen/src/display.rs

```diff
@@ -813,3 +813,74 @@ pub fn parse_doc_for_param(doc: &str, param: &str) -> String {
     .trim()
     .to_string()
 }
+
+/// Takes a doc string from a Q# callable and returns all parameter names and their descriptions
+/// as a Vec<(String, String)>, extracted from the # Input(s) section.
+#[must_use]
+pub fn parse_doc_for_all_params(doc: &str) -> Vec<(String, String)> {
```

Comment:
We're ending up with quite a few different parsers in this file to extract various parts of doc comments. I wonder if we shouldn't just have one good doc comment parser that turns a doc comment into a strongly typed struct that every other helps can call and take the parts it needs from. (Maybe not for this PR)